src/Migrations/M033MigrateSocialMediaTwitterBlockToEmbedBlock.php

```diff
@@ -0,0 +1,286 @@
+        }
+        return '';
+    }
+
+    /**
+     * Transform a Social Media block into a group of blocks.
+     * This group contains a header for the Social Media block title,
+     * a paragraph for the Social Media block description,
+     * and a final block (Embed) for the Twitter/X embed.
+     *
+     * @param array $block - A parsed Social Media block.
+     * @param string $source - The source of the Social Media block (Twitter or X).
+     * @param string $social_media_url - The Social Media URL.
+     * @return array - The transformed block.
+     */
+    private static function transform_blocks(array $block, string $social_media_url): array
+    {
+        $block_title = array_key_exists("title", $block['attrs']) ? $block['attrs']['title'] : null;
+        $block_description = array_key_exists("description", $block['attrs']) ? $block['attrs']['description'] : null;
+
+        $block = [];
+        $block['blockName'] = Utils\Constants::BLOCK_GROUP;
+        $block['attrs']['metadata']['name'] = 'Twitter/X Group';
+
+        $title = $block_title ? Utils\Functions::create_block_heading(
+            ['level' => 2],
+            $block_title
+        ) : null;
+
+        $description = $block_description ? Utils\Functions::create_block_paragraph(
+            [],
+            $block_description
+        ) : null;
+
+        $block['innerBlocks'] = [];
+        $block['innerBlocks'][0] = $title;
+        $block['innerBlocks'][1] = $description;
+        $block['innerBlocks'][3] = self::transform_block_to_embed($block, $social_media_url);
+
+        // IMPORTANT: DO NOT MODIFY THIS FORMAT!
+        $block['innerHTML'] =
+        '<div class="wp-block-group">
+
+
+
+
+
+        </div>';
+
+        // IMPORTANT: DO NOT MODIFY THIS FORMAT!
+        $block['innerContent'] = array (
+            0 => '
+        <div class="wp-block-group">',
+            1 => null,
+            2 => '
+        ',
+            3 => null,
+            4 => '
+        ',
+            5 => null,
+            6 => '
+        ',
+            7 => null,
+            8 => '</div>
+        ',
+        );
+
+        return $block;
+    }
+
+    /**
+     * Transform a Social Media block into an embed block.
+     *
+     * @param array $block - A parsed Social Media block.
+     * @param string $social_media_url - The Social Media URL.
+     * @return array - The transformed block.
+     */
+    private static function transform_block_to_embed(array $block, string $social_media_url): array
+    {
+        $type = 'rich';
+        $provider = self::SOURCE_TWITTER;
+
+        // IMPORTANT: DO NOT MODIFY THIS FORMAT!
+        $html_content = '<figure class="wp-block-embed is-type-' . $type . ' is-provider-' . $provider . ' wp-block-embed-' . $provider . ' wp-embed-aspect-16-9 wp-has-aspect-ratio"><div class="wp-block-embed__wrapper">
+        ' . $social_media_url . '
+        </div></figure>';
+
+        $block['blockName'] = Utils\Constants::BLOCK_EMBED;
```

Comment:
Can you please add a function to the Utils/Functions.php file to make the creation of Embed blocks reusable?